### PR TITLE
Implement stop query functionality.

### LINKED ISSUE #2129: Implement a way to stop the query

### PR DESCRIPTION
**Pull request by bkyryliuk**
Fixes: https://github.com/airbnb/superset/issues/2129
Tested on staging and locally

@mistercrunch @ascott ^^

### CODE REVIEW

File: superset/assets/javascripts/SqlLab/components/SqlEditor.jsx [hunk 1]
```diff
@@ -83,7 +83,7 @@ class SqlEditor extends React.PureComponent {
     this.props.actions.setActiveSouthPaneTab('Results');
   }
   stopQuery() {
```
Comment (ascott) [inline superset/assets/javascripts/SqlLab/components/SqlEditor.jsx]: should we rename this to `cancelQuery` as well to match the action name?

File: superset/assets/javascripts/SqlLab/components/SqlEditor.jsx [hunk 2]
```diff
@@ -83,7 +83,7 @@ class SqlEditor extends React.PureComponent {
...
-    this.props.actions.stopQuery(this.props.latestQuery);
```
Comment (ascott) [inline superset/assets/javascripts/SqlLab/components/SqlEditor.jsx]: is the `stopQuery` action used elsewhere? if not we should remove it in `superset/assets/javascripts/SqlLab/actions.js` as well

File: superset/assets/javascripts/SqlLab/actions.js [hunk 3]
```diff
@@ -142,6 +142,25 @@ export function runQuery(query) {
...
+    const cancelQueryUrl = '/superset/stop_query/';
+    const cancelQueryRequest = { client_id: query.id };
+    $.ajax({
+      type: 'POST',
+      dataType: 'json',
+      url: cancelQueryUrl,
+      data: cancelQueryRequest,
+      success(results) {
+        if (!query.runAsync) {
+          dispatch(querySuccess(query, results));
```
Comment (ascott) [inline superset/assets/javascripts/SqlLab/actions.js]: i'm curious why we want to call `querySuccess` here. wouldn't we want to show feedback that the query has been cancelled? `querySuccess` is used to show the fetched results.

File: superset/assets/javascripts/SqlLab/actions.js [hunk 4]
```diff
@@ -142,6 +142,25 @@ export function runQuery(query) {
   };
 }
 
+export function cancelQuery(query) {
+  return function (dispatch) {
+    dispatch(stopQuery(query));
```
Comment (ascott) [inline superset/assets/javascripts/SqlLab/actions.js]: what is the difference between `cancelQuery` and `stopQuery`?

**Comment by ascott** [comment]
would be great to see a gif of how this works on the front end.

File: superset/assets/javascripts/SqlLab/actions.js (hunk 4 above)
Comment (bkyryliuk) [inline superset/assets/javascripts/SqlLab/actions.js]: renamed to the postStopQuery

File: superset/assets/javascripts/SqlLab/actions.js (hunk 3 above)
Comment (bkyryliuk) [inline superset/assets/javascripts/SqlLab/actions.js]: great catch, I need to dispatch stopQuery

File: superset/assets/javascripts/SqlLab/components/SqlEditor.jsx (hunk 2 above)
Comment (bkyryliuk) [inline superset/assets/javascripts/SqlLab/components/SqlEditor.jsx]: postStopQuery uses it

File: superset/assets/javascripts/SqlLab/components/SqlEditor.jsx (hunk 1 above)
Comment (bkyryliuk) [inline superset/assets/javascripts/SqlLab/components/SqlEditor.jsx]: named it postStopQuery

**Comment by bkyryliuk** [comment]
@ascott  - no visible changes on the UI.
deployed: https://superset-staging-b.d.musta.ch/superset/sqllab

File: superset/assets/javascripts/SqlLab/actions.js [hunk 5]
```diff
@@ -142,6 +142,24 @@ export function runQuery(query) {
   };
 }
 
+export function postStopQuery(query) {
+  return function (dispatch) {
+    const stopQueryUrl = '/superset/stop_query/';
+    const cancelQueryRequest = { client_id: query.id };
```
Comment (ascott) [inline superset/assets/javascripts/SqlLab/actions.js]: can we call  this`stopQueryRequestData` for consistency?

File: superset/assets/javascripts/SqlLab/actions.js [hunk 6]
```diff
@@ -142,6 +142,24 @@ export function runQuery(query) {
...
+    $.ajax({
+      type: 'POST',
+      dataType: 'json',
+      url: stopQueryUrl,
+      data: cancelQueryRequest,
+      success() {
+        if (!query.runAsync) {
```
Comment (ascott) [inline superset/assets/javascripts/SqlLab/actions.js]: why don't we want to stop the query if it's async?

Comment (bkyryliuk) [inline superset/assets/javascripts/SqlLab/actions.js]: async pulls the status from the server - different flow.

**Comment by ascott** [comment]
lgtm

**Comment by bkyryliuk** [comment]
thanks @ascott !

**Pull request merged**